source/manuals/programming-languages/java.html.md.erb

```diff
@@ -101,8 +113,10 @@ Keep in mind that improvements to the Java standard library mean that some exter
 
 When using external libraries, favour those which complement the Java standard library. For example, be wary of any library which introduces a new type that replicates the functionality of an existing type in the Java standard library without implementing the same interfaces (for example, a list type which does not implement `java.util.List`).
 
+Make sure any external library you use is appropriate for your purposes and avoid relying on internal implementation details of external libraries. If your IDE’s code completion suggests a method from an external library, make sure it’s a supported part of the library’s defined API.
```

Comment:
thoughts on adding in some detail about accidentally relying on transitive dependencies?

Reply:
I did half-write a sentence about this but since it wasn’t actually something we discussed in the meeting, I left it out for now.